src/components/CIPipelineN/CustomInputOutputVariables.tsx

```diff
@@ -0,0 +1,322 @@
+import React, { useContext, useState } from 'react'
+import { ReactComponent as Close } from '../../assets/icons/ic-close.svg'
+import { ReactComponent as Add } from '../../assets/icons/ic-add.svg'
+import { ReactComponent as Equal } from '../../assets/icons/ic-variable-equal.svg'
+import {
+    FormType,
+    PluginVariableType,
+    RefVariableType,
+    TaskFieldDescription,
+    VariableFieldType,
+    VariableType,
+} from '../ciPipeline/types'
+import CustomInputVariableSelect from './CustomInputVariableSelect'
+import { ciPipelineContext } from './CIPipeline'
+import { ReactComponent as AlertTriangle } from '../../assets/icons/ic-alert-triangle.svg'
+import ReactSelect from 'react-select'
+import { tempMultiSelectStyles } from './ciPipeline.utils'
+import Tippy from '@tippyjs/react'
+import { Option } from '../v2/common/ReactSelect.utils'
+import { OptionType } from '../app/types'
+
+function CustomInputOutputVariables({ type }: { type: PluginVariableType }) {
+    const {
+        formData,
+        setFormData,
+        selectedTaskIndex,
+        activeStageName,
+        calculateLastStepDetail,
+        formDataErrorObj,
+    }: {
+        formData: FormType
+        setFormData: React.Dispatch<React.SetStateAction<FormType>>
+        selectedTaskIndex: number
+        activeStageName: string
+        calculateLastStepDetail: (
+            isFromAddNewTask: boolean,
+            _formData: FormType,
+            activeStageName: string,
+            startIndex?: number,
+        ) => {
+            index: number
+            calculatedStageVariables: Map<string, VariableType>[]
+        }
+        formDataErrorObj: object
+    } = useContext(ciPipelineContext)
+    const formatOptions: OptionType[] = ['STRING', 'BOOL', 'NUMBER', 'DATE'].map((format) => ({
+        label: format,
+        value: format,
+    }))
+    const addVariable = (): void => {
+        const _formData = { ...formData }
+        const id =
+            _formData[activeStageName].steps[selectedTaskIndex].inlineStepDetail[VariableFieldType[type]]?.reduce(
+                (prev, current) => (prev.id > current.id ? prev : current),
+                {
+                    id: 0,
+                },
+            ).id + 1
+        const newCondition = {
+            id: id,
+            name: '',
+            value: '',
+            format: formatOptions[0].label,
+            description: '',
+            defaultValue: '',
+            refVariableUsed: false,
+            variableType: RefVariableType.NEW,
+            refVariableStepIndex: 0,
+            refVariableName: '',
+        }
+        if (!_formData[activeStageName].steps[selectedTaskIndex].inlineStepDetail[VariableFieldType[type]]) {
+            _formData[activeStageName].steps[selectedTaskIndex].inlineStepDetail[VariableFieldType[type]] = []
```

Comment:
same here - move `_formData[activeStageName].steps[selectedTaskIndex].inlineStepDetail[VariableFieldType[type]]` to some variable.